src/api/app/views/webui2/webui/package/_revision_diff_detail_buttons.html.haml

```diff
@@ -7,9 +7,10 @@
   href: "#revision_details_#{index + 1}",
   title: 'Go down to the next diff' }
     %i.fas.fa-chevron-down
-  - link_options = { project: project, package: package, filename: filename, rev: calculate_revision_on_state(revision, state) }
-  - if !@linkinfo && (has_content && !Package.is_binary_file?(filename) && !filename.include?('/'))
-    = link_to package_view_file_path(link_options),
+  - revision = previous_revision(revision) if state == 'deleted'
```

Comment:
Why not using the `@last_rev` variable? instead of the `previous_revision` method? :thinking:

Reply:
That said, I would prefer this in a separate PR, since it's not clear yet if and how this can be solved differently.